packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -135,34 +135,74 @@ function buildAndRun(args: Flags) {
     packageName,
   );
   const adbPath = getAdbPath();
-  if (args.deviceId) {
+  let devices;
+
+  if (!args.deviceId) {
+    devices = adb.getDevices(adbPath);
+  }
+
+  if (
+    devices &&
+    devices.length > 1 &&
+    devices.filter(Boolean).length > 1 &&
+    args.interactive
+  ) {
+    devices = await chooseDevice(devices);
+  }
+
+  if (args.deviceId || (devices && devices.length === 1)) {
     return runOnSpecificDevice(
       args,
       cmd,
       packageNameWithSuffix,
       packageName,
       adbPath,
+      devices,
     );
-  } else {
+  } else if (devices && devices.length > 1) {
     return runOnAllDevices(
```

Comment:
While we are at it, why we can't just `devices.forEach(d => runOnSpecificDevice(....))` but need a separate function? Could be potentially a good area for refactor.